Firebase Push Notifications.md

```diff
@@ -0,0 +1,88 @@
+# Firebase push notifications
+
+Table of contents:
+
+1. [General](#general)
+2. [Setup](#setup)
+3. [Development approaches](#development-approaches)
+4. [Choosing right development approach](#choosing-development-approach)
+
+## General
+In this chapter we will explain few approaches for push notifications implementations and theirs PROS and CONS.
+
+There are many services which provides us with push notifications, but we are using Firebase because of its simplicity, reliability and because it's free.
+
+Firebase Cloud Messaging handles the routing and delivery of push notifications to targeted devices.
+
+Largest size of one message is 4KB.
+
+It is possible to send push notification to only one device, group of devices or to topics.
+
+Before development, read chapters about [Firebase Cloud Messaging](https://firebase.google.com/docs/cloud-messaging/), and [app server development](https://firebase.google.com/docs/cloud-messaging/server).
+
+## Setup
+1. DevOps has to create new Firebase project.
+
+2. Server key from Firebase project settings should be copied to Vault. Server key authorizes your app server on Google services.
+
+3. Add [fcm gem](https://github.com/spacialdb/fcm) to your project. Fcm gem has methods for sending requests to [Firebase](https://firebase.google.com/docs/cloud-messaging/send-message).
```

Comment:
I would say "The Fcm gem serves as an SDK for communicating with the Firebase Service"